boefjes/boefjes/plugins/kat_rdns/main.py

```diff
@@ -0,0 +1,25 @@
+from typing import List, Tuple, Union
+
+import dns
+from dns.resolver import Answer
+
+from boefjes.config import settings
+from boefjes.job_models import BoefjeMeta
+
+
+def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
+    """return results to normalizer."""
+    ip = boefje_meta.arguments["input"]["address"]
+
+    try:
+        resolver = dns.resolver.Resolver()
+        resolver.nameservers = [settings.remote_ns]
+        reverse_ip = dns.reversename.from_address(ip)
+        answer: Answer = resolver.resolve(reverse_ip, "PTR")
+        result = f"RESOLVER: {answer.nameserver}\n{answer.response}"
+        return [(set(), result)]
+    except dns.resolver.NXDOMAIN:
+        return [(set(), "NXDOMAIN")]
+    except (dns.resolver.Timeout, dns.resolver.NoAnswer):
+        pass
+    return [(set(), "No PTR record found.")]
```

Comment:
"No PTR record found" sounds like NXDOMAIN to me, e.g. we get a response that there is no PTR record. I think in case of timeout/noanswer the result should be something like "No answer"

Reply:
I agree, If we don't get a timeout, but do get a NoAnswer, there might be two tings:
A there is no record for this IP (ar at least, there is no PRT record for the requested host / reversed IP)
B, the server who's authoritative for this range is mis configured / broken.
I'm not sure we can tell the difference?